pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -333,30 +336,31 @@ func (r *Reconciler) sendLog(ctx context.Context, o results.Object) error {
 			zap.String("name", o.GetName()),
 		)
 
-		go func() {
-			err := r.streamLogs(ctx, o, logType, logName)
-			if err != nil {
-				logger.Errorw("Error streaming log",
-					zap.String("namespace", o.GetNamespace()),
-					zap.String("kind", o.GetObjectKind().GroupVersionKind().Kind),
-					zap.String("name", o.GetName()),
-					zap.Error(err),
-				)
-			}
-			logger.Debugw("Streaming log completed",
+		err = r.streamLogs(ctx, o, logType, logName)
+		if err != nil {
+			logger.Errorw("Error streaming log",
 				zap.String("namespace", o.GetNamespace()),
 				zap.String("kind", o.GetObjectKind().GroupVersionKind().Kind),
 				zap.String("name", o.GetName()),
+				zap.Error(err),
 			)
-		}()
+		}
+		logger.Debugw("Streaming log completed",
+			zap.String("namespace", o.GetNamespace()),
+			zap.String("kind", o.GetObjectKind().GroupVersionKind().Kind),
+			zap.String("name", o.GetName()),
+		)
+
 	}
 
 	return nil
 }
 
 func (r *Reconciler) streamLogs(ctx context.Context, o results.Object, logType, logName string) error {
-	logger := logging.FromContext(ctx)
-	logsClient, err := r.resultsClient.UpdateLog(ctx)
+	streamCtx, streamCancel := context.WithTimeout(ctx, 5*time.Minute)
```

Comment:
if we go with the broader using of context timeout within the dynamic reconciler loop, we would get rid of this particular child context with timeout